contracts/p1/Distributor.sol

```diff
@@ -201,6 +233,33 @@ contract DistributorP1 is ComponentP1, IDistributor {
         require(rTokenDist != 0 || rsrDist != 0, "no distribution defined");
     }
 
+    function _validateDistributionTable() internal view {
+        uint256 dLen = destinations.length();
+        DAOFeeRegistry daoFeeRegistry = main.daoFeeRegistry();
+        require(destinations.length() > 0, "no destinations");
+
+        RevenueTotals memory revTotals;
+        uint24 daoTotals;
+
+        for (uint256 i = 0; i < dLen; ++i) {
+            address dest = destinations.at(i);
+            RevenueShare storage share = distribution[dest];
+            revTotals.rTokenTotal += share.rTokenDist;
+            revTotals.rsrTotal += share.rsrDist;
+
+            if (dest == DAO_FEE) {
+                daoTotals += share.rsrDist;
+            }
+        }
+
+        require(
```

Comment:
i think there's a better way to do this.  instead of having stRSR set the distribution %'s and hoping they match what the daoFeeRegistry is expecting, we could have `_setDistribution` call the daoFeeRegistry for the % cut that it is requiring from both rsrDist and rTokenDist for this particular RToken, and then calculate what that works out to, and set the daoFee dist accordingly.

for efficiency, we can remove the singular `setDistribution` and only have 'setDistributions`, so that the final daoFee calculation can be done once everything else is set.

Reply:
We don't want to remove a function, but we'd usually use `setDistributions`. The reason we're passing the values is because (1) they can choose to pay more than required; and (2) the calculation is gas heavy to be run at the end so better to let it be computed off-chain.

Reply:
@akshatmittal what I said earlier about not wanting to change the interface was a bad reason given this is 4.0.0

Something else we didn't consider is that under the currently-implemented design, if the DAO fee is raised, there has to be a push to get RTokens to call their `setDistributions()` functions. They may not want to, though. 


Possible other design: public() function executed at end of `setDistributions()` and externally callable only by veRSR

Reply:
Let me rephrase, I agree with don't need to keep the same interface here, but I see no harm in keeping the old interface and also adding a new one.

I think I like the idea that veRSR distribution is _dynamic_ and always comes from the Fee Registry in real time, so it's never actually added to the distributions but rather something that's always present regardless of the entire table.

Reply:
I like this direction

Reply:
yup this makes sense to me